sdk/program/src/epoch_rewards.rs

```diff
@@ -44,7 +44,7 @@ impl EpochRewards {
     pub fn distribute(&mut self, amount: u64) {
         assert!(self.distributed_rewards.saturating_add(amount) <= self.total_rewards);
 
-        self.distributed_rewards.add_assign(amount);
+        self.distributed_rewards = self.distributed_rewards.saturating_add(amount);
```

Comment:
The lint seems unnecessary here, but the change to saturating add is innocuous. I don't think we need to do the addition multiple times, though
How about this:
```
pub fn distribute(&mut self, amount: u64) {
    let new_distributed_rewards = self.distributed_rewards.saturating_add(amount);
    assert!(new_distributed_rewards <= self.total_rewards);
    self.distributed_rewards = new_distributed_rewards;
}
```

Reply:
I like it.

Reply:
thank you for the insight! yeah, good catch! just fixed it! [34f81e7](https://github.com/anza-xyz/agave/pull/2493/commits/34f81e7e653d33e64ae1d220721936dfac980137)